apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using Google.Cloud.Spanner.V1.Logging;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+
+// ReSharper disable PrivateFieldCanBeConvertedToLocalVariable
+// ReSharper disable NotAccessedField.Local
+// ReSharper disable EmptyDestructor
+
+namespace Google.Cloud.Spanner.V1
+{
+    /// <summary>
+    /// Provides streaming access to a Spanner SQL query that automatically retries, handles
+    /// chunking and recoverable errors.
+    /// </summary>
+    public sealed class ReliableStreamReader : IDisposable
+    {
+        private AsyncServerStreamingCall<PartialResultSet> _currentCall;
+        private readonly object _sync = new object();
+        private readonly SpannerClient _spannerClient;
+        private readonly ExecuteSqlRequest _request;
```

Comment:
the instance remains the same and is set in the ctor.  We will be changing the resume token quite often as we attempt to reconnect after failures.